router/handle_observability.go

```diff
@@ -147,3 +149,33 @@ func (rt *Handle) pipelineDelayStats(partition string, first, last *jobsdb.JobT)
 	stats.Default.NewTaggedStat("pipeline_delay_min_seconds", stats.GaugeType, stats.Tags{"destType": rt.destType, "partition": partition, "module": "router"}).Gauge(lastJobDelay)
 	stats.Default.NewTaggedStat("pipeline_delay_max_seconds", stats.GaugeType, stats.Tags{"destType": rt.destType, "partition": partition, "module": "router"}).Gauge(firstJobDelay)
 }
+
+// eventOrderDebugInfo provides some debug information for the given orderKey in case of a panic.
+// Top 100 job statuses for the given orderKey are returned.
+func (rt *Handle) eventOrderDebugInfo(orderKey string) (res string) {
+	defer func() {
+		if r := recover(); r != nil {
+			res = fmt.Sprintf("panic in EventOrderDebugInfo: %v", r)
+		}
+	}()
```

Comment:
Is this going to catch the `panic` and not propagate it anymore? So the next time we got an out of order rudder-server won't crash? I thought you would save the data and re-issue the `panic`.

Reply:
My bad, this is for the local `panic`.